libbeat/beat/beat.go

```diff
@@ -195,6 +216,9 @@ func (b *Beat) Run() {
 }
 
 // Stop calls the beater Stop action.
+// It can happen that this function is called more then once.
 func (beat *Beat) Stop() {
+	logp.Info("Stopping Beat")
 	beat.BT.Stop()
+	close(beat.done)
```

Comment:
this method is called by signal handler. Maybe has only the close for `beat.done` and call stop + wait for shutdown complete using a wait group in Run method starting the go routine.